lib/json-schema/attributes/allof.rb

```diff
@@ -29,7 +29,7 @@ def self.validate(current_schema, data, fragments, processor, validator, options
         end
 
         if !valid || !errors.empty?
-          message = "The property '#{build_fragment(fragments)}' of type #{data.class} did not match all of the required schemas"
+          message = "The property '#{build_fragment(fragments)}' of type #{type_of_data(data)} did not match all of the required schemas"
```

Comment:
Can you explain why this is necessary? (I assume something to do with the Fixnum/BigNum/Integer changes?)